### PR TITLE
refactor: enhance draw text functiones for better flexibility

### PR DESCRIPTION
**Pull request by GoldO-dev**
## Description

- Changed `scale` parameter in `qbx.drawText3d` to be a `vec2` with a default value of `vec2(0.35, 0.35)`, while maintaining backwards compatibility with the previous version
- Introduced `disableDropShadow` and `disableOutline` parameters in `qbx.drawText2d`
- Introduced `enableDropShadow` and `enableOutline` parameters in `qbx.drawText3d`
- Added a warning for users using a number value as the scale parameter in `qbx.drawText3d`, encouraging the use of a `vec2` instead.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: modules/lib.lua
Comment (Manason) [inline modules/lib.lua:388]: Let's remove this. I don't think we need to force people to update their scripts over this.

File: modules/lib.lua
Comment (Manason) [inline modules/lib.lua:348]: Why does 2D use disable while 3D uses enable? 2D defaults these features to on if unset whereas 3D does not.

**Review by Manason — CHANGES_REQUESTED** [review]

File: modules/lib.lua
Comment (GoldO-dev) [reply modules/lib.lua:348]: > Why does 2D use disable while 3D uses enable? 2D defaults these features to on if unset whereas 3D does not.

The reason I did it like that is because before I changed stuff there was a outline and drop shadow on the 2d text by default and none on the 3d text by default. So I wanted to keep it like that.  

This is the original code for drawText2d as you can see by default there is a outline and drop shadow
```lua
    function qbx.drawText2d(params)
        local text = params.text
        local coords = params.coords
        local scale = params.scale or 0.35
        local font = params.font or 4
        local color = params.color or vec4(255, 255, 255, 255)
        local width = params.width or 1.0
        local height = params.height or 1.0
        SetTextScale(scale, scale)
        SetTextFont(font)
        SetTextColour(math.floor(color.r), math.floor(color.g), math.floor(color.b), math.floor(color.a))
        SetTextDropShadow()
        SetTextOutline()
        SetTextCentre(true)
        BeginTextCommandDisplayText('STRING')
        AddTextComponentSubstringPlayerName(text)
        EndTextCommandDisplayText(coords.x - width / 2, coords.y - height / 2 + 0.005)
    end
```

File: modules/lib.lua
Comment (GoldO-dev) [reply modules/lib.lua]: Done

File: modules/lib.lua
Comment (GoldO-dev) [reply modules/lib.lua:347]: Do you want me to change it? Or can I keep it the way I've done?

Comment (GoldO-dev) [reply modules/lib.lua:347]: @Manason am I allowed to mark this conversation as resolved?

Comment (Manason) [reply modules/lib.lua:347]: Yeah let's default it either way for both consistently.

Comment (GoldO-dev) [reply modules/lib.lua:347]: Not sure I totally understand, what you mena by that. Can you elaborate?

Comment (Manason) [reply modules/lib.lua:347]: Let's either default shadow to on and have the disable param, or default it to off and have the enable param. But I don't want them to have different params.

Comment (GoldO-dev) [reply modules/lib.lua:347]: Okay, what would you prefer?

Comment (Manason) [reply modules/lib.lua:347]: Probably default off with an optional enable param

Comment (GoldO-dev) [reply modules/lib.lua:347]: Okay, will change it quick. I also have a updated public documentation fork, where I will change the stuff on too. And make a pull request on the docs repository.

**Comment by GoldO-dev** [comment]
@Manason all requested changes has been made.

I've also made a pull request to the docs repository that updates the docs to match the changes made in this pull request. You can take a look at it [here](https://github.com/Qbox-project/qbox-project.github.io/pull/43)

**Review by Manason — APPROVED** [review]

File: modules/lib.lua
Comment (Manason) [inline modules/lib.lua:368]: Can you add back integer as well? I think we'll support both going forward rather than deprecate the original integer param.

**Review by Manason — APPROVED** [review]

File: modules/lib.lua
Comment (GoldO-dev) [reply modules/lib.lua:368]: The way it's done at the moment it supports a integer as well. It just turns it into a vector2 under the hood since it make the code a bit easier to read. 

Would you like me to do it like this?
```lua
---@field scale? integer | vector2 default: `0.35`
```

Since the code already handles integers, I don't see a need to change any of the code in the actual function. Correct me if I'm wrong.

Comment (Manason) [reply modules/lib.lua:368]: You are correct.

Comment (GoldO-dev) [reply modules/lib.lua]: @Manason the latest commit should fix it. Can you check to confirm it's made how you wanted?

**Review by Manason — APPROVED** [review]

**Comment by GoldO-dev** [comment]
@Manason just a quick question, do I need a second reviewer to review before it can be merged? Or can it be merged by just you giving it a review?

**Comment by Manason** [comment]
> @Manason just a quick question, do I need a second reviewer to review before it can be merged? Or can it be merged by just you giving it a review?

A second reviewer is required.

**Review by mafewtm — APPROVED** [review]

**Pull request merged**
